### ISSUE TITLE
question regarding the use of logsoftmax in S1.Viewpoint

### ISSUE THREAD

**Issue by ZhengYi0310** [leoshine/Spherical_Regression#11]
Hi,
   thanks very much for providing the code !
   I was looking though the regEulerNet.py and having a question regarding the use of logsoftmax in:
   https://github.com/leoshine/Spherical_Regression/blob/d19bc2f6f52982d4d58f5ddabe4231381d7facd7/S1.Viewpoint/model/regEulerNet.py#L208-L210

   the logprob_ccss computed here will be used to compute the negtive dot product with the ground truth sin ** 2 and cos ** 2:
   https://github.com/leoshine/Spherical_Regression/blob/d19bc2f6f52982d4d58f5ddabe4231381d7facd7/S1.Viewpoint/model/regEulerNet.py#L228
   https://github.com/leoshine/Spherical_Regression/blob/d19bc2f6f52982d4d58f5ddabe4231381d7facd7/S1.Viewpoint/model/regEulerNet.py#L280-L288

   GT[tgt] come from the dataset_naiReg.py:
   https://github.com/leoshine/Spherical_Regression/blob/d19bc2f6f52982d4d58f5ddabe4231381d7facd7/S1.Viewpoint/lib/datasets/dataset_naiReg.py#L222-L224
   Which seem consistent with eqn(9) in the paper.

   However the use of logsoftmax doesn't seem consistent with eqn (9), shouldn't it be softmax ?
   
   Thanks in advance !

**Comment by leoshine** [comment]
Hi,

To the right of Eq(9) in the paper, you may find the actual loss we use:     H(Y^2, P^2) = - \sum y_i^2  log(p_i^2)
Yes, Eq(9) specifies the computation of p_i^2, however, the code directly computes log(p_i^2) to plug into the loss function.

The practical reason for using logsoftmax is that logsoftmax is numerically stable than softmax.
